### PR TITLE
fix: Correct selection of menu item in Navigation drawer

### LINKED ISSUE #1383: Incorrect display of selected item in Navigation Drawer
**Describe the bug**
Select Check-In tab in Navigation Drawer then view activity log of any attendee and then press back to land up in Attendee List Fragment. Now, again view the activity log for any attendee and see the navigation drawer. The tab selected will be Dashboard instead of Check-In.

**Expected behavior**
The selcted tab should be Check-In whenever user is in Check-In or it's child fragments.

**Screenshots**
ezgif-6-ce7c6418d905

**Would you like to work on the issue?**

- [ ] Yes
- [x] No

### PR DESCRIPTION
**Pull request by kush-mish**
Fixes #1383 

Checklist:

- [ ] I have checked for PMD and check-style issues 
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream development branch.

Note: This bug was not afflicting only the Attendees fragment having the Activity Log as a subfragment but, every other fragment having subfragments.

Changes: Correct menu item is selected in Navigation drawer even when the back button is pressed from a subfragment.

Screenshot after the change:
alt text

### CODE REVIEW

**Comment by kush-mish** [comment]
@iamareebjamal, please help me with the test

**Comment by iamareebjamal** [comment]
Tests are failing. Fix them

**Comment by kush-mish** [comment]
The testBack() function in FragmentNavigator.java is asserting that the dashboard should be active after pressing back, which seems implausible since when inside a subfragment pressing back will lead to the fragment and not the dashboard.
https://github.com/fossasia/open-event-orga-app/blob/c116c294992905348b3bbc1b6a057fe31e382b3c/app/src/test/java/com/eventyay/organizer/core/main/FragmentNavigatorTest.java#L111-L129
That's why line 126 is causing error.

**Comment by iamareebjamal** [comment]
True, add a test case for that and fix this issue

File: app/src/test/java/com/eventyay/organizer/core/main/FragmentNavigatorTest.java [hunk 1]
```diff
@@ -123,7 +124,9 @@ public void testBack() {
 
         fragmentNavigator.back();
 
-        assertTrue(fragmentNavigator.isDashboardActive());
+        if (fragmentManager.getBackStackEntryCount() == 1)
+            assertTrue(fragmentNavigator.isDashboardActive());
+        else assertFalse(fragmentNavigator.isDashboardActive());
```
Comment (iamareebjamal) [inline app/src/test/java/com/eventyay/organizer/core/main/FragmentNavigatorTest.java]: No if statements in tests

**Pull request merged**